sponsorship_compassion/model/res_partner.py

```diff
@@ -119,9 +119,7 @@ def create_contract(self):
             'view_mode': 'form',
             'res_model': 'recurring.contract',
             'target': 'current',
-            'context': self.env.context.with_context({
-                'default_partner_id': self,
-                'default_type': 'S', })
+            'context': context

```

Comment:
Set `context` key with the dictionnary used in `with_context` above, and remove line 111